timemachine/cpp/src/bd_exchange_move.cu

```diff
@@ -162,15 +165,18 @@ void BDExchangeMove<RealType>::move(
     curandErrchk(templateCurandUniform(cr_rng_translations_, d_translations_.data, d_translations_.length));
     curandErrchk(templateCurandUniform(cr_rng_samples_, d_sample_noise_.data, d_sample_noise_.length));
     curandErrchk(templateCurandUniform(cr_rng_mh_, d_mh_noise_.data, d_mh_noise_.length));
-    for (int step = 0; step < steps_per_move_; step++) {
+    size_t step = 0;
+    // For the first pass just set the value to zero on the host
+    p_noise_offset_.data[0] = 0;
+    while (p_noise_offset_.data[0] < num_proposals_per_move_) {
```

Comment:
should add a brief but clear description about the overall algorithm at the start of the while loop

Reply:
Added description here, let me know what you think: https://github.com/proteneer/timemachine/pull/1274/commits/fc62abe77506e217f422b5bf5a340e19e4c3acee